e2e/commands_test.go

```diff
@@ -124,179 +125,174 @@ maintainers:
 	assert.Assert(t, fs.Equal(dirName, manifest))
 
 	// validate metadata with JSON Schema
-	AssertCommand(t, dockerApp, "validate", testAppName)
+	icmd.RunCommand(dockerApp, "validate", testAppName).Assert(t, icmd.Success)
 
 	// test single-file init
-	args = []string{
-		"init",
-		"tac",
-		"-c",
-		dir.Join(internal.ComposeFileName),
-		"-d",
-		"my cool app",
+	icmd.RunCommand(dockerApp, "init", "tac",
+		"-c", dir.Join(internal.ComposeFileName),
+		"-d", "my cool app",
 		"-m", "bob",
 		"-m", "joe:joe@joe.com",
 		"-s",
-	}
-	AssertCommand(t, dockerApp, args...)
+	).Assert(t, icmd.Success)
 	defer os.Remove("tac.dockerapp")
 	appData, err := ioutil.ReadFile("tac.dockerapp")
 	assert.NilError(t, err)
 	golden.Assert(t, string(appData), "init-singlefile.dockerapp")
 	// Check various commands work on single-file app package
-	AssertCommand(t, dockerApp, "inspect", "tac")
-	AssertCommand(t, dockerApp, "render", "tac")
+	icmd.RunCommand(dockerApp, "inspect", "tac").Assert(t, icmd.Success)
+	icmd.RunCommand(dockerApp, "render", "tac").Assert(t, icmd.Success)
 }
 
-func TestDetectAppBinary(t *testing.T) {
-	dockerApp, _ := getDockerAppBinary(t)
+func TestDetectApp(t *testing.T) {
 	// cwd = e2e
-	AssertCommand(t, dockerApp, "inspect")
+	dir := fs.NewDir(t, "detect-app-binary",
+		fs.WithDir("helm.dockerapp", fs.FromDir("testdata/helm.dockerapp")),
+		fs.WithDir("render",
+			fs.WithDir("app1.dockerapp", fs.FromDir("testdata/render/envvariables/my.dockerapp")),
+			fs.WithDir("app2.dockerapp", fs.FromDir("testdata/render/envvariables/my.dockerapp")),
+		),
+	)
+	defer dir.Remove()
 	cwd, err := os.Getwd()
 	assert.NilError(t, err)
-	assert.NilError(t, os.Chdir("helm.dockerapp"))
+	assert.NilError(t, os.Chdir(dir.Path()))
```

Comment:
This *could* (or more accurately **should**) be removed and using `icmd.Cmd{…, Dir: dir.Path())`, …